src/plugins/controls/public/options_list/embeddable/options_list_embeddable.tsx

```diff
@@ -322,6 +322,12 @@ export class OptionsListEmbeddable extends Embeddable<OptionsListEmbeddableInput
           },
           this.abortController.signal
         );
+      if (rejected) {
+        // This prevents a rejected request (which can happen, for example, when a user types a search string too quickly)
+        // from prematurely setting loading to `false` and updating the suggestions to show "No results"
+        return;
```

Comment:
To test this, remove this early return, throttle your network, and type something super fast into the search bar - you'll notice that you'll see a flash of the loading state, then you'll see the `"No results"` message which will hang, before finally seeing your expected results: